js/classes/camera.js

```diff
@@ -35,8 +30,20 @@ class Camera {
             }
             return z;
         };
-        const x = median(DIMENSION.MIN_X, DIMENSION.MAX_X - Camera.WIDTH, CHAD.pos.x - Camera.WIDTH / 2);
-        const y = median(DIMENSION.MIN_Y, DIMENSION.MAX_Y - Camera.HEIGHT, CHAD.pos.y - Camera.HEIGHT / 2);
+        // This is where the Camera is if it can be perfectly centered on CHAD.
+        const centeredOnChad = new Vector(
+            CHAD.pos.x + (PapaChad.SIZE.x / 2) - (Camera.SIZE.x / 2),
+            CHAD.pos.y + (PapaChad.SIZE.y / 2) - (Camera.SIZE.y / 2));
+
+        // This is where the Camera is if it is at the lower right corner of the Zone.
+        const maxPos = new Vector(ZONE.MAX_PT.x - Camera.SIZE.x, ZONE.MAX_PT.y - Camera.SIZE.y);
```

Comment:
There are 3 x values and 3 y values which must be considered. I grouped them into 3 points (the three obvious ones - min, max, and centered on Chad. The resulting camera.pos does not have to MATCH one of these three points; however, it's x value must be one of the vectors x positions, and its y must be one of the y values. 

Indeed, I could have combined the 3 X and 3 Y variables together any way, and ended up with the same answer. I agree it is a slightly unconventional way to use them, but I don't think it's improper at all - in my mind, it's a quick and easy way to use less verbose code, and keep the namespace clean.

Reply:
That's fair. My comment mainly comes from the fact that you group each of the min, max, and centered x's and y's into Vectors but otherwise treat them independently. It certainly is not very important, but I was curious what your thoughts were. Thanks!

Reply:
@tclaar i added some comments and used more of the static Vector methods instead of so much subtraction in the constructors and whatnot. I think its good tho.